### PR TITLE
feat: add vite generators. Drop in nx web replacement

### PR DESCRIPTION
**Pull request by Jordan-Hall**
Feature: Vite alternative to @nrwl/web (webpack) 
Closes: #357

- [x] Create vite generators
- [x] Create React Generators
- [x] Add documentation
- [x] Add E2E testing
- [x] move solid:lib to `vite:package`

### CODE REVIEW

**Comment by Jordan-Hall** [comment]
@DominikPieper  any idea why its failing on linux CI only?

File: packages/docs-docusaurus/docs/react/overview.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+npm install -D @nxext/react
+```
+
+> Warning: unlike the webpack version provided by [NX](https://nx.dev). Our vite builds does not support images as components. Instead when you import images `import logo from 'logo.svg` you will get the image URL to be used inside of the image tag `<img src={logo}/>`
+
+## Executors / Builders
+
+Our React applications are built using the executors from the [@nxext/vite](../vite/overview.md) plugin.
+
+- [build](../vite/executors/build) - Builds a web components application
```
Comment (DominikPieper) [inline packages/docs-docusaurus/docs/react/overview.md:33]:
```suggestion
- [build](../vite/executors/build.md) - Builds a web components application
```

File: packages/docs-docusaurus/docs/react/overview.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+- [dev](..vite/executors/serve) - Builds and serves a web application
```
Comment (DominikPieper) [inline packages/docs-docusaurus/docs/react/overview.md:34]:
```suggestion
- [dev](../vite/executors/serve.md) - Builds and serves a web application
```

File: packages/docs-docusaurus/docs/react/overview.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+- [package](../vite/executors/package) - Bundles artifacts for a buildable library that can be distributed as an NPM package.
```
Comment (DominikPieper) [inline packages/docs-docusaurus/docs/react/overview.md:35]:
```suggestion
- [package](../vite/executors/package.md) - Bundles artifacts for a buildable library that can be distributed as an NPM package.
```

File don't exist yet

File: packages/docs-docusaurus/docs/react/overview.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+## Generators
+
+We have adapted the @nrwl/react plugin to create to create the scaffolding for the vite builds. For components and redux we still recommend using @nrwl/react
+
+- [application](./generators/application) - Create an React application
```
Comment (DominikPieper) [inline packages/docs-docusaurus/docs/react/overview.md:41]:
```suggestion
- [application](./generators/application.md) - Create an React application
```

File: packages/docs-docusaurus/docs/react/overview.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+- [library](./generators//library) - Create an React library
```
Comment (DominikPieper) [inline packages/docs-docusaurus/docs/react/overview.md:42]:
```suggestion
- [library](./generators//library.md) - Create an React library
```

**Comment by DominikPieper** [comment]
@Jordan-Hall do you want to do every task in this PR? I'd say let's do the cypress and storybook changes in it's own PRs. What do you think?

**Comment by Jordan-Hall** [comment]
> @Jordan-Hall do you want to do every task in this PR? I'd say let's do the cypress and storybook changes in it's own PRs. What do you think?

100% Cypress and storybook scope been removed. I did want a drop in replacement release, but then realise the scope would be too big.

**Comment by DominikPieper** [comment]
The drop-in replacement is a good idea, but I'd just say not in one PR :-) I mostly split things up into multiple PRs. The build and test times are better than as well :-D e2e-tests are so time consuming to run

File: packages/docs-docusaurus/docs/react/overview.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+```
+
+> Warning: unlike the webpack version provided by [NX](https://nx.dev). Our vite builds does not support images as components. Instead when you import images `import logo from 'logo.svg` you will get the image URL to be used inside of the image tag `<img src={logo}/>`
+
+## Executors / Builders
+
+Our React applications are built using the executors from the [@nxext/vite](../vite/overview.md) plugin.
+
+- [build](../vite/executors/build.md) - Builds a web components application
+- [dev](..vite/executors/serve.md) - Builds and serves a web application
```
Comment (DominikPieper) [inline packages/docs-docusaurus/docs/react/overview.md:34]:
```suggestion
- [dev](../vite/executors/serve.md) - Builds and serves a web application
```

**Comment by Jordan-Hall** [comment]
@DominikPieper  Im struggling with the e2e sorry. I'm getting the following errors

`packages/react/src/generators/init/init.ts:9:35 - error TS2307: Cannot find module '@nxext/vite' or its corresponding type declarations.`

This only fails in the e2e tests. I've got the same e2e as solid JS so im a bit confused

**Comment by DominikPieper** [comment]
How did your beforeAll looks like?

**Comment by Jordan-Hall** [comment]
> How did your beforeAll looks like?

```
beforeAll(() => {
    ensureNxProjectWithDeps('@nxext/react', 'dist/packages/react', [
      ['@nxext/vite', 'dist/packages/vite'],
    ]);
  });
```

**Comment by DominikPieper** [comment]
Could you check the react e2e test dir? Seems like you copied everything into the tests folder as well

**Comment by Jordan-Hall** [comment]
> Could you check the react e2e test dir? Seems like you copied everything into the tests folder as well

I did just copy after realizing we using tools/e2e executor. I''ll have to come back to this tonight now as I need to start work for a client today. So will be back on this in about 10 hours. Sorry

**Comment by DominikPieper** [comment]
@Jordan-Hall all fine. the tools/e2e executor isn't used anymore and it's replaced just with the ensureNxProjectWithDeps

File: e2e/react-e2e/project.json [hunk 7]
```diff
@@ -0,0 +1,16 @@
+{
+  "root": "e2e/react-e2e",
+  "projectType": "application",
+  "sourceRoot": "e2e/react-e2e/src",
+  "targets": {
+    "e2e": {
+      "executor": "./tools/executors/e2e:e2e",
+      "options": {
+        "targets": ["vite:build", "react:build"],
```
Comment (DominikPieper) [inline e2e/react-e2e/project.json:9]:
```suggestion
        "targets": ["react:build", "vite:build"],
```
The order matters @Jordan-Hall

**Comment by Jordan-Hall** [comment]
Thanks for the help on fixing that test issue. This should now resolve the last E2E test for both vite and react.

**Comment by Jordan-Hall** [comment]
@DominikPieper ready to be merged now :)

**Comment by DominikPieper** [comment]
@Jordan-Hall oh wow, great. I'll take a look later today and merge it :-) I already looked at most of the changes and features and some of the vite features you added for React are the foundation I need for Svelte

**Comment by Jordan-Hall** [comment]
@DominikPieper Just pushed another useful change :) Migrated Solidjs library builds to vite:package and upgraded the assets copy for libs

**Review by DominikPieper — APPROVED** [review]

**Pull request merged**